ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -117,6 +118,49 @@ end = struct
     mode := next_mode
 
   let current_mode () = !mode
+
+  let is_ipv6_enabled () = !ipv6_enabled
+end
+
+module Client_certificate_auth_server = struct
+  let management_server = ref None
+
+  let must_be_running ~__context ~mgmt_enabled =
+    let pool = Helpers.get_pool ~__context in
+    mgmt_enabled
+    && Pool_role.is_master ()
+    && Db.Pool.get_client_certificate_auth_enabled ~__context ~self:pool
+
+  let configure_port cmd =
+    let cmd' = if cmd then "open" else "close" in
+    let _ =
+      Helpers.call_script
+        !Xapi_globs.firewall_port_config_script
+        [cmd'; string_of_int Constants.https_port_clientcert]
+    in
+    ()
+
+  let start () =
+    if !management_server = None then (
+      let sock_path = Xapi_globs.unix_domain_socket_clientcert in
+      Unixext.mkdir_safe (Filename.dirname sock_path) 0o700 ;
+      Unixext.unlink_safe sock_path ;
+      let domain_sock = Xapi_http.bind (Unix.ADDR_UNIX sock_path) in
+      Http_svr.start Xapi_http.server domain_sock ;
+      configure_port true ;
+      management_server := Some domain_sock
+    )
+
+  let stop () =
```

Comment:
From what I can tell, if a master changes to a slave this will be called when the toolstack reboots. This means the port gets closed as expected

Similarly `stop` is called when disabling client auth, so the port is closed in this case too :+1:

Reply:
I think that is true, yes.